core/src/main/java/software/amazon/awssdk/http/pipeline/stages/ApplyUserAgentStage.java

```diff
@@ -31,20 +32,38 @@
 public class ApplyUserAgentStage implements MutableRequestToRequestPipeline {
 
     private final LegacyClientConfiguration config;
+    private final String clientName;
+    private final String clientType;
 
-    public ApplyUserAgentStage(HttpClientDependencies dependencies) {
+    public ApplyUserAgentStage(HttpClientDependencies dependencies,
+                               String clientName,
+                               String clientType) {
```

Comment:
Having a client name of ApacheAsync would work but I switched to separate values as then we could at least guarantee that Sync vs Async would be correct as we hardcode it into AmazonHttpClient and AmazonAsyncHttpClient. That way even if the client name metric is not always perfect because it is up to the implementor, sync vs async will always be correct.